### PR TITLE
240. Search a 2D matrix II

### PR DESCRIPTION
**Pull request by MateuszSapala**

### CODE REVIEW

**Review by baranskim — APPROVED** [review]

**Review by barbara-k97 — APPROVED** [review]

**Comment by ArsenEloglian** [comment]
@baranskim & @barbara-k97 
was @MateuszSapala in our lessons today?

**Comment by barbara-k97** [comment]
Yes.

**Comment by ArsenEloglian** [comment]
So why does 0320cac41d0ff11b476b075d2f03e9410f81f450 commited twice and be approved by you if it's already merged? Why does PR contain files which should be ignored?

Kindly ask you to be precise during the review process.

File: AttendanceList.txt [hunk 1]
```diff
@@ -44,6 +44,7 @@ Mateusz Sapala
   21.10.2022
   28.10.2022
   04.11.2022
+  18.11.2022
```
Comment (ArsenEloglian) [inline AttendanceList.txt:47]: already exist in code

File: MateuszSapala/240.Search_a_2D_Matrix_II/pom.xml [hunk 2]
```diff
@@ -0,0 +1,74 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ArsenEloglian) [inline MateuszSapala/240.Search_a_2D_Matrix_II/pom.xml:1]: file should be ignored by gitignore

File: MateuszSapala/240.Search_a_2D_Matrix_II/pom.xml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (ArsenEloglian) [inline MateuszSapala/240.Search_a_2D_Matrix_II/pom.xml:2]: file should be ignored by gitignore

File: MateuszSapala/240.Search_a_2D_Matrix_II/src/main/kotlin/SearchMatrix.kt [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                        return true
+                    }
+                    i++
+                }
+                j--
+            }
+            return false
+        }
+    }
+}
```
Comment (ArsenEloglian) [inline MateuszSapala/240.Search_a_2D_Matrix_II/src/main/kotlin/SearchMatrix.kt:75]: looking good, just remember about endlines

File: MateuszSapala/240.Search_a_2D_Matrix_II/src/test/kotlin/SearchMatrix.kt [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+                    intArrayOf(1, 2, 3, 4, 5),
+                    intArrayOf(6, 7, 8, 9, 10),
+                    intArrayOf(11, 12, 13, 14, 15),
+                    intArrayOf(16, 17, 18, 19, 20),
+                    intArrayOf(21, 22, 23, 24, 25),
+                ), 15, true
+            ),
+        )
+    }
+}
```
Comment (ArsenEloglian) [inline MateuszSapala/240.Search_a_2D_Matrix_II/src/test/kotlin/SearchMatrix.kt:75]: the same here

**Review by ArsenEloglian — CHANGES_REQUESTED** [review]

File: AttendanceList.txt (hunk 1 above)
Comment (MateuszSapala) [reply AttendanceList.txt]: Unnecessary commit removed

File: MateuszSapala/240.Search_a_2D_Matrix_II/pom.xml (hunk 2 above)
Comment (MateuszSapala) [reply MateuszSapala/240.Search_a_2D_Matrix_II/pom.xml]: Without this file you won't run project. If this bothers you so much, I will commit only sollution code files from now on

**Review by barbara-k97 — APPROVED** [review]

**Review by Flehmango — APPROVED** [review]
lgtm

**Review by ArsenEloglian — APPROVED** [review]

**Pull request merged**